kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/util/KubernetesUtils.java

```diff
@@ -161,6 +166,35 @@ public static PropertySource secretAsPropertySource(Secret secret) {
         return excludesFilter;
     }
 
+    /**
+     * @param client the {@link KubernetesClient}
+     * @param podLabelKeys the list of labels inside a pod
+     * @param namespace in the configuration
+     * @return the filtered labels of the current pod
+     */
+    public static Map<String, String> getPodLabels(KubernetesClient client, List<String> podLabelKeys, String namespace) {
```

Comment:
A better name would be `computePodLabels`. `getPodLabels` seems like a getter, and also the same as `KubernetesConfiguration.AbstractKubernetesConfiguration.getPodLabels()`